extensions/amp-script/0.1/amp-script.js

```diff
@@ -390,27 +392,62 @@ export class AmpScript extends AMP.BaseElement {
       );
     }
     const allowMutation =
-      // Hydration is always allowed.
-      phase != Phase.MUTATING ||
       this.isMutationAllowedByFixedSize() ||
       this.isMutationAllowedByUserGesture();
 
-    if (allowMutation) {
-      this.vsync_.mutate(flush);
-      return;
-    }
+    this.vsync_.mutate(() => {
+      const disallowedTypes = flush(allowMutation);
+      // Count the number of mutations dropped by type.
+      const errors = map();
+      disallowedTypes.forEach(type => {
+        errors[type] = errors[type] + 1 || 1;
+      });
+      // Emit an error message for each mutation type, including count.
+      Object.keys(errors).forEach(type => {
```

Comment:
nit: `Object.entries` would give you both key and value, for a more concise loop.

Reply:
Not safe to use unfortunately. https://caniuse.com/#feat=object-entries

Looks like we ban it in conformance-config.textproto too.